core/src/main/java/hudson/PluginManager.java

```diff
@@ -446,6 +478,40 @@ public void run(Reactor reactor) throws Exception {
         }});
     }
 
+    @Extension
+    public final static PluginManagerAdministrativeMonitor NOTICE = new PluginManagerAdministrativeMonitor();
```

Comment:
@daniel-beck 

> It's not clear why this needs to run so early during initialization. This is only for UI use, right?

Yes, but only for the moment...

> So why not wait until plugins and extensions are done loading?

Because later on I would like to fix JENKINS-21486, and then I will need to move the code here again.

https://issues.jenkins-ci.org/browse/JENKINS-21486

Anyway, if for the moment you prefer I move this elsewhere I am fine... For the moment, I think this an important feature and I am mostly interested in merging and releasing this.